### PR TITLE
Improve Importer TryToUseMethodDefs and TryToUseFieldDefs options 2

### PR DESCRIPTION
**Pull request by wwh1004**
Obfuscator may rename many methods/fields into same name then TryResolveMethod/TryResolveField will return inconsistent method/field.

Previous PR contains some problems and I fixed it. In previous PR you mentioned 'privatescope' visibility. I check some documents and find it is visible in the same module and can't be referenced in MemberRef table. I think these code won't get error when encountering 'privatescope' visibility. Should it be handled specially?

### CODE REVIEW

**Comment by CreateAndInject** [comment]
It's better use `AssemblyNameComparer` or `SigComparer` to determine if is the same Module/Assembly

File: src/DotNet/Importer.cs [hunk 1]
```diff
@@ -508,6 +516,8 @@ public Importer(ModuleDef module, ImporterOptions options, GenericParamContext g
 			}
 		}
 
+		bool IsThisModule(Module module2) => module.Name == module2.Name && IsThisAssembly(module2);
```
Comment (wtfsck) [inline src/DotNet/Importer.cs:519]: `SigComparer` uses `module2.ScopeName` and it should be case insensitive comparison, see `bool Equals(IModule a, Module b) {`.

**Review by wtfsck — CHANGES_REQUESTED** [review]

**Comment by wtfsck** [comment]
> Previous PR contains some problems and I fixed it. In previous PR you mentioned 'privatescope' visibility. I check some documents and find it is visible in the same module and can't be referenced in MemberRef table. I think these code won't get error when encountering 'privatescope' visibility. Should it be handled specially?

I don't think it's necessary to check for `privatescope` here.

**Comment by wtfsck** [comment]
> It's better to use `AssemblyNameComparer` or `SigComparer` to determine if is the same Module/Assembly

True, but in this case only the name is compared and there's no public API that only compares the name.

File: src/DotNet/Importer.cs (hunk 1 above)
Comment (wwh1004) [reply src/DotNet/Importer.cs]: Done.

**Review by wtfsck — APPROVED** [review]

**Comment by wtfsck** [comment]
Thanks, merged!

**Pull request merged**